### PR TITLE
[APP-6278] Assemble OTel headers dynamically from entity ID + live API token

### PR DESCRIPTION
**Pull request by andriykislitsyn**
## CHANGES

OTel OTLP auth headers now use the live API token instead of values frozen in pulumi_config.json at deploy time.

## Checklist
- [ ] Reviewed [guideline](https://datarobot.atlassian.net/wiki/spaces/BUZOK/pages/7305920528/REVIEW+BEFORE+COMMIT+Working+with+agentic+starter+application+and+its+components)
- [ ] Updated Changelog
- [ ] New tools in `src/datarobot_genai/drtools/` require to add `integration` tests in `tests/drmcp/integration/` and `acceptance` tests in `tests/drmcp/acceptance/` (this is a MUST)

---

> [!NOTE]
> **Low Risk**
> Telemetry auth configuration only; explicit headers and env precedence are preserved, with tests covering edge cases.
> 
> **Overview**
> **OTel OTLP auth headers now use the live API token** instead of values frozen in `pulumi_config.json` at deploy time.
> 
> `MCPServerConfig` adds a `before` validator on `otel_exporter_otlp_headers`: when the header string is unset, it builds `x-datarobot-entity-id` + `x-datarobot-api-key` from `OTEL_ENTITY_ID` (or `otel_entity_id`) and the current `DATAROBOT_API_TOKEN`. Explicit `OTEL_EXPORTER_OTLP_HEADERS` is unchanged.
> 
> The DRAgent CLI `_bridge_pulumi_otel_env()` follows the same pattern at startup: it reads `OTEL_ENTITY_ID` and the endpoint from `pulumi_config.json`, skips if headers are already in the environment, and otherwise assembles headers from entity ID + `DATAROBOT_API_TOKEN`, with a fallback to legacy baked `OTEL_EXPORTER_OTLP_HEADERS` in the JSON when entity ID or token is missing.
> 
> Release **0.15.111**; unit tests cover assembly, precedence, and fallbacks for both MCP config and the CLI bridge.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8c08169228e525969dc811cfa1f83229146de507. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/datarobot_genai/drmcp/core/config.py [hunk 1]
```diff
@@ -284,6 +285,21 @@ def _get_default_otel_endpoint() -> str:
...
+        if v:
+            return v
+        entity_id = (info.data or {}).get("otel_entity_id", "")
+        api_token = os.environ.get("DATAROBOT_API_TOKEN", "")
+        if entity_id and api_token:
+            return (
+                f"x-datarobot-entity-id={entity_id},"
+                f"x-datarobot-api-key={api_token}"
+            )
+        return v
```
[the remark this answers is not in the document]
Comment (andriykislitsyn) [reply src/datarobot_genai/drmcp/core/config.py:308]: The validator ordering concern is technically valid in theory, but doesn't apply in practice, OTEL_EXPORTER_OTLP_HEADERS is never delivered as a runtime parameter. The platform injects it as a plain env var.

**Review by carsongee — APPROVED** [review]

**Review by rabih-datarobot — APPROVED** [review]
LGTM

**Review by akshoop — APPROVED** [review]
LGTM!
